business/istio_certs.go

```diff
@@ -0,0 +1,143 @@
+package business
+
+import (
+	"fmt"
+	"sync"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+)
+
+type IstioCertsService struct {
+	k8s           kubernetes.ClientInterface
+	businessLayer *Layer
+}
+
+type certConfig struct {
+	SecretName string   `yaml:"secretName"`
+	DNSNames   []string `yaml:"dnsNames"`
+}
+
+type istioConfig struct {
+	Certificates []certConfig `yaml:"certificates"`
+}
+
+const (
+	IstioDefaultCASecret string = "istio-ca-secret"
+	UserProvidedCASecret string = "cacerts"
```

Comment:
confirm these are the names in OSSM, too.

And if they can be different (OSSM? or canary deployments?) these should be configurable in our configmap somewhere (in [the same way the name of the configmap is configurable](https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L543)).

Reply:
@jmazzitelli thanks for your comments.

I confirm that secrets names in OSSM are the same.

Regarding canary deployments, not an expert but I did a test upgrading from 1.9.6 to 1.10.3 seeing that there is not a new secret created after the upgrade, it seems to be using the same "istio-ca-secret".

I will continue with this new strategy and will update the message as you proposed.

Reply:
With the new message:

![certs_chiron_new_msg](https://user-images.githubusercontent.com/1286393/129051431-54b08f54-3ef3-48ff-aafa-5cd661fea6bb.png)

Reply:
Sorry, I'm late to the review, I see several colleagues are working together to review this one.
A first comment from my side, in this screen capture, I'd suggest a minor change in the design:

"From dns.example1-service-account secret"
(Warning icon) Access Denied + Tooltip with the original message

In that case, if this list grow it can handle it better, or if you have a list of several certificates without access, you can benefit of this design (note that other cards are designed in that way).

Reply:
@lucasponce something like the following?

![certs_new_error](https://user-images.githubusercontent.com/1286393/129983020-7b8e99d0-096e-4e97-b018-cc35e76a54cb.png)

Reply:
@leandroberetta yes +1